tt_metal/programming_examples/contributed/vecadd/vecadd.cpp

```diff
@@ -0,0 +1,215 @@
+// SPDX-FileCopyrightText: © 2024 Martin Chang
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "common/core_coord.h"
+#include "tt_metal/host_api.hpp"
+#include "common/bfloat16.hpp"
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <random>
+#include <string_view>
+#include <vector>
+
+using namespace tt::tt_metal;
+
+using CoreSpec = std::variant<CoreCoord, CoreRange, CoreRangeSet>;
+
+std::shared_ptr<Buffer> MakeBuffer(Device *device, uint32_t size, uint32_t page_size, bool sram)
+{
+    InterleavedBufferConfig config{
+        .device= device,
+        .size = size,
+        .page_size = page_size,
+        .buffer_type = (sram ? BufferType::L1 : BufferType::DRAM)
+    };
+    return CreateBuffer(config);
+}
+
+// Allocate a buffer on DRAM or SRAM. Assuming the buffer holds BFP16 data.
+// A tile on Tenstorrent is 32x32 elements, given us using BFP16, we need 2 bytes per element.
+// Making the tile size 32x32x2 = 2048 bytes.
+// @param device: The device to allocate the buffer on.
+// @param n_tiles: The number of tiles to allocate.
+// @param sram: If true, allocate the buffer on SRAM, otherwise allocate it on DRAM.
+std::shared_ptr<Buffer> MakeBufferBFP16(Device *device, uint32_t n_tiles, bool sram)
+{
+    constexpr uint32_t tile_size = 2 * (32 * 32);
```

Comment:
I think it would be good to use the defined values here instead of hardcoding the numbers.
Ex. `constexpr uint32_t tile_size = sizeof(bfloat16) * (TILE_HEIGHT * TILE_WIDTH);` or `constexpr uint32_t tile_size = tt_metal::detail::TileSize(tt::DataFormat::Float16_b);` (first one is probably preferred if you want to show how the size is derived)